examples/apps/simple_wallet/workload/simple_wallet_execute_io.cpp

```diff
@@ -0,0 +1,187 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <string.h>
+
+#include "types.h"
+#include "jsonvalue.h"
+#include "json_utils.h"
+#include "error.h"
+#include "tcf_error.h"
+#include "enclave_utils.h"
+#include <parson.h>
+
+#include "simple_wallet_execute_io.h"
+#include "trusted_iohandler.h"
+#include "crypto_iohandler.h"
+
+#define LEDGER_FILE "/tmp/account_ledger.json"
+
+// Placeholder to store all ledger entries. Useful when updating
+// given account in the ledger.
+std::string g_ledgerDetails;
+
+// Create Wallet ledger file to store details of transactions in encrypted format
+void CreateWalletLedger() {
+    uint32_t status = ExecuteFileWrite("");
+    if (status == 0) {
+        Log(TCF_LOG_INFO, "Successfully created wallet ledger file %s",
+            LEDGER_FILE);
+    } else {
+        Log(TCF_LOG_INFO, "Failed to create wallet ledger file %s",
+            LEDGER_FILE);
+    }
+}
+
+// Get wallet balance for given account from json serialized ledger
+std::string GetWalletBalanceFromLedger(std::string ledgerData,
+    std::string accountName) {
+    // Parse account_ledger.json data
+    JsonValue parsed(json_parse_string(ledgerData.c_str()));
+    tcf::error::ThrowIfNull(parsed.value,
+        "failed to parse the work order request, badly formed JSON");
+
+    JSON_Object* ledger_object = json_value_get_object(parsed);
+    tcf::error::ThrowIfNull(ledger_object, "Missing JSON object in ledger");
+    Log(TCF_LOG_INFO, "Getting balance for the account from ledger object: %s",
+        accountName);
+    std::string balance = GetJsonStr(ledger_object, accountName.c_str(),
+        nullptr);
+    Log(TCF_LOG_INFO, "GetWalletBalanceFromLedger::Got balance for account %s",
+        accountName);
```

Comment:
It depends. In case of Deposit, if account is not present we need to create a new account.
For other transactions, if account is not present error is thrown (already covered).